src/panels/SessionRecording/RecordSession.tsx

```diff
@@ -0,0 +1,147 @@
+import { useState } from 'react';
+import { Checkbox, Group, TextInput, Title } from '@mantine/core';
+
+import { useOpenSpaceApi, useSubscribeToSessionRecording } from '@/api/hooks';
+import { useAppDispatch, useAppSelector } from '@/redux/hooks';
+import { updateSessionRecordingSettings } from '@/redux/sessionrecording/sessionRecordingSlice';
+
+import { RecordButton } from './Buttons/RecordButton';
+import { StopRecordingButton } from './Buttons/StopRecordingButton';
+import { RecordingState } from './types';
+
+export function RecordSession() {
+  const [filenameRecording, setFilenameRecording] = useState('');
+  const [filenameState, setFilenameState] = useState({
+    invalid: false,
+    errorMessage: ''
+  });
+  const { overwriteFile } = useAppSelector((state) => state.sessionRecording.settings);
+  const [showOverwriteCheckbox, setShowOverwriteCheckbox] = useState(overwriteFile);
+
+  const recordingState = useSubscribeToSessionRecording();
+  const luaApi = useOpenSpaceApi();
+
+  const dispatch = useAppDispatch();
+  const fileList = useAppSelector((state) => state.sessionRecording.files);
+
+  function isIdle(): boolean {
+    return recordingState === RecordingState.Idle;
+  }
+
+  function isRecordingState(): boolean {
+    return (
+      recordingState === RecordingState.Idle ||
```

Comment:
Why is Idle part of this check... ? Also, I'm honestly a bit confused of what the point of this `isRecordingState` function is